### ISSUE TITLE
Add: Setup Nginx SSL with minio server

### ISSUE THREAD

**Issue by schenkp** [minio/cookbook#65]
I use minio server as docker container together with nginx-proxy container. All works fine when using non SSL connection, but with SSL a simple: _mc ls configname_ does not work.

For example, I get a:

_mc: <ERROR> Unable to list folder. The provided 'x-amz-content-sha256' header does not match what was computed._

On the other hand, when i provide a bucket name like in:

_mc ls <configname>/bucket_

i get results. It seems that the auth header is not correct for some reason. Is there anything that would be needed in the headers to be passed to the server when using Nginx in SSL mode, proxying HTTP to minio? An example for "Setup Nginx SSL with minio server" would be very much appreciated.

**Comment by harshavardhana** [comment]
Here is a document on configuring nginx proxy  https://docs.minio.io/docs/setup-nginx-proxy-with-minio . Also can you paste you 

- nginx version
- minio server version
- mc version

??

**Comment by koolhead17** [comment]
Hello @schenkp this should be fixed via https://github.com/minio/cookbook/commit/ab021efc108ac9f8812212396e06ab53ec058bc2  Please confirm so that we can close this. Thanks again. 👍

**Comment by schenkp** [comment]
@koolhead: yes, this works when using non SSL.

In my setup I configured nginx for HTTPS proxy to but minio server with HTTP. What i have seen is that minio-go.v2 does not sign and leaves shaHeader with UNSIGNED-PAYLOAD for X-Amz-Content-Sha256 when using secure. Minio server then complains with mc: Unable to list folder. The provided 'x-amz-content-sha256' header does not match what was computed.

So i looked into minio-go-v2 to see what happens with the X-Amz-Content-Sha256, and I changed the "if !c.secure" line into "if c.secure":

// newRequest - instantiate a new HTTP request for a given method.
func (c Client) newRequest(method string, metadata requestMetadata) (req *http.Request, err error) {

	// Set sha256 sum only for non anonymous credentials.
	if !c.anonymous {
		// set sha256 sum for signature calculation only with
		// signature version '4'.
		if c.signature.isV4() {
			shaHeader := "UNSIGNED-PAYLOAD"

=>			if c.secure {  // was if ! c.secure

				if metadata.contentSHA256Bytes == nil {
					shaHeader = hex.EncodeToString(sum256([]byte{}))
				} else {
					shaHeader = hex.EncodeToString(metadata.contentSHA256Bytes)
				}
			}
			req.Header.Set("X-Amz-Content-Sha256", shaHeader)
		}
	}

With that NGINX HTTPS proxy to minio server HTTP works, but also MC to MINIO (both HTTP) works.
